### PR TITLE
feat(rms/policy): add new policy assignment resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new policy assignment resource support for RMS service.
Some parameters have the following constraints:
- If the `period` is configured, it means that the evaluation is performed periodically.
- If the `policy_filter` is configured, it means that the evaluation is performed on the specified resources through the filter.
- If neither parameter is configured, it means that the evaluation is performed on all resources under the account.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new policy assignment resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rms' TESTARGS='-run=TestAccPolicyAssignment_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rms -v -run=TestAccPolicyAssignment_ -timeout 360m -parallel 4
=== RUN   TestAccPolicyAssignment_basic
=== PAUSE TestAccPolicyAssignment_basic
=== RUN   TestAccPolicyAssignment_period
=== PAUSE TestAccPolicyAssignment_period
=== RUN   TestAccPolicyAssignment_custom
=== PAUSE TestAccPolicyAssignment_custom
=== CONT  TestAccPolicyAssignment_basic
=== CONT  TestAccPolicyAssignment_custom
=== CONT  TestAccPolicyAssignment_period
--- PASS: TestAccPolicyAssignment_period (143.01s)
--- PASS: TestAccPolicyAssignment_custom (281.59s)
--- PASS: TestAccPolicyAssignment_basic (299.84s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rms       299.923s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/rms' TESTARGS='-run=TestAccPolicyAssignment_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rms -v -run=TestAccPolicyAssignment_ -timeout 360m -parallel 4
=== RUN   TestAccPolicyAssignment_basic
=== PAUSE TestAccPolicyAssignment_basic
=== RUN   TestAccPolicyAssignment_period
=== PAUSE TestAccPolicyAssignment_period
=== RUN   TestAccPolicyAssignment_custom
=== PAUSE TestAccPolicyAssignment_custom
=== CONT  TestAccPolicyAssignment_basic
=== CONT  TestAccPolicyAssignment_custom
=== CONT  TestAccPolicyAssignment_period
--- PASS: TestAccPolicyAssignment_period (135.05s)
--- PASS: TestAccPolicyAssignment_custom (281.73s)
--- PASS: TestAccPolicyAssignment_basic (302.03s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rms       302.250s
```

File: docs/resources/rms_policy_assignment.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
+---
+subcategory: "Resource Management Service (RMS)"
+---
+
+# huaweicloud_rms_policy_assignment
+
+Manages a (API) custom response resource within HuaweiCloud.
```
Comment (niuzhenguo) [inline docs/resources/rms_policy_assignment.md]: Manages a RMS resource...

File: docs/resources/rms_policy_assignment.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the policy assignment is located.  
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `name` - (Required, String) Specifies the name of the policy assignment.  
+  The valid length is limited from can contain `1` to `64`, only letters, digits, hyphens (-) and underscores (_) are
```
Comment (niuzhenguo) [inline docs/resources/rms_policy_assignment.md]: get rid of `can contain`

File: docs/resources/rms_policy_assignment.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  allowed. The name must start with a letter.
+
+* `description` - (Optional, String) Specifies the description of the policy assignment, which contain maximum of
+  512 characters.
+
+* `type` - (Optional, String) Specifies the type of the policy assignment.  
+  The valid values are as follows:
+  + **builtin**
+  + **custom**
```
Comment (niuzhenguo) [inline docs/resources/rms_policy_assignment.md]: better to mention the Default type here.

File: docs/resources/rms_policy_assignment.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+* `policy_definition_id` - (Optional, String) Specifies the ID of the built-in policy definition.  
+  This parameter and `custom_policy` are alternative.
+
+* `period` - (Optional, String) Specifies the period of the policy assignment.  
+  The valid values are as follows:
+  + **One_Hour**
+  + **Three_Hours**
+  + **Six_Hours**
+  + **Twelve_Hours**
+  + **TwentyFour_Hours**
```
Comment (niuzhenguo) [inline docs/resources/rms_policy_assignment.md:114]: ditto.

File: docs/resources/rms_policy_assignment.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  type        = "custom"
+  name        = var.policy_assignment_name
+  description = "The ECS instances that do not conform to the custom function logic are considered non-compliant."
+  status      = "Enabled"
+
+  custom_policy {
+    function_urn = "${var.function_urn}:${var.function_version}"
+    auth_type    = "agency"
+    auth_value   = {
+      "agency_name": "\"${var.rms_admin_trust_agency}\""
```
Comment (ShiChangkuo) [inline docs/resources/rms_policy_assignment.md]:
```suggestion
      agency_name: "\"${var.rms_admin_trust_agency}\""
```

the key of a map  does not need quotation marks, please update them in docs and acceptance

File: huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go [hunk 6]
```diff
@@ -0,0 +1,578 @@
...
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The region where the policy assignment is located.",
+			},
+			"domain_id": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go]: the domain_id is a constant value for users, and we cat get it from config, so I think we should not define it in schema

File: huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go [hunk 7]
```diff
@@ -0,0 +1,578 @@
...
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go]: please check whether RMS is global, if yes, we can delete this parameter.

File: docs/resources/rms_policy_assignment.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/rms_policy_assignment.md]: no default value, and the value is depend on the configuration of the parameter `policy_definition_id` and the parameter `custom_policy`.

File: docs/resources/rms_policy_assignment.md (hunk 4 above)
Comment (Lance52259) [reply docs/resources/rms_policy_assignment.md:114]: no default value, if omitted, means that the evaluation is performed on all resources under the account.

File: huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go]: sure, this parameter has been removed.

File: docs/resources/rms_policy_assignment.md (hunk 5 above)
Comment (Lance52259) [reply docs/resources/rms_policy_assignment.md]: all quotation marks have been deleted

File: huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go (hunk 7 above)
Comment (Lance52259) [reply huaweicloud/services/rms/resource_huaweicloud_rms_policy_assignment.go]: the RMS service is the global service, the parameter `region` should be deleted.

File: docs/resources/rms_policy_assignment.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/rms_policy_assignment.md]: fixed.

File: docs/resources/rms_policy_assignment.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/rms_policy_assignment.md]: fixed.

File: docs/resources/rms_policy_assignment.md (hunk 3 above)
Comment (niuzhenguo) [reply docs/resources/rms_policy_assignment.md]: ok

File: docs/resources/rms_policy_assignment.md (hunk 4 above)
Comment (niuzhenguo) [reply docs/resources/rms_policy_assignment.md:114]: ok

**Pull request merged**